custom_components/yeelight_bt/light.py

```diff
@@ -291,6 +299,15 @@ async def async_turn_on(self, **kwargs: int) -> None:
 
         # if ATTR_EFFECT in kwargs:
         #    self._effect = kwargs[ATTR_EFFECT]
+        
+        if ColorMode.HS in self.supported_color_modes and ATTR_HS_COLOR in kwargs:

```

Comment:
Isn't this already implemented at line 266?

Reply:
yes you're right, I moved it out to debug an issue I was having and totally forgot about it. I've pushed an update